auth/auth.go

```diff
@@ -0,0 +1,27 @@
+package auth
+
+type key string
+
+const (
+	// The current auth configuration that has been set by any auth middleware and
+	// that will prevent from being set more than once.
+	CURRENT_AUTH_CONFIG key = "currentAuthConfig"
+)
+
+// An HTTP-specific authentication scheme that sends an arbitrary
+// auth value in a header or query string parameter.
+// As described in the Smithy documentation:
+// https://github.com/awslabs/smithy/blob/main/smithy-model/src/main/resources/software/amazon/smithy/model/loader/prelude.smithy
+type HttpAuthDefinition struct {
```

Comment:
this seems to be `HttpApiKeyAuthDefinition`, no? I see the same name used in the corresponding TypeScript PR which is merged in.

Reply:
This is a generic auth definition used for both API Key and [Bearer token](https://smithy.io/1.0/spec/core/auth-traits.html#httpbearerauth-trait) (ideally it should be used for for [httpBasicAuth](https://smithy.io/1.0/spec/core/auth-traits.html#httpbasicauth-trait) and [httpDigestAuth](https://smithy.io/1.0/spec/core/auth-traits.html#httpdigestauth-trait) traits as well).